src/pages/[gatewayUID]/details.tsx

```diff
@@ -1,32 +1,72 @@
 /* eslint-disable react/jsx-props-no-spreading */
-import type { GetServerSideProps, NextPage } from "next";
+import { useState, useEffect } from "react";
+import type { NextPage } from "next";
 import { useRouter } from "next/router";
 import { ParsedUrlQuery } from "querystring";
-import { useState } from "react";
-import { changeGatewayName } from "../../api-client/gateway";
+import { changeGatewayName, useGateway } from "../../api-client/gateway";
 import GatewayDetails from "../../components/elements/GatewayDetails";
 import { LoadingSpinner } from "../../components/layout/LoadingSpinner";
-import { services } from "../../services/ServiceLocator";
 import { getErrorMessage } from "../../constants/ui";
-import { ERROR_CODES } from "../../services/Errors";
-
+import { useNodes } from "../../api-client/node";
 import GatewayDetailViewModel from "../../models/GatewayDetailViewModel";
 import { getGatewayDetailsPresentation } from "../../components/presentation/gatewayDetails";
 
-type GatewayDetailsData = {
-  viewModel: GatewayDetailViewModel;
-  err?: string;
-};
+interface GatewayDetailsQueryInterface extends ParsedUrlQuery {
+  gatewayUID: string;
+}
 
-const GatewayDetailsPage: NextPage<GatewayDetailsData> = ({
-  viewModel,
-  err,
-}) => {
+const GatewayDetailsPage: NextPage = () => {
+  const [viewModel, setViewModel] = useState<GatewayDetailViewModel>({});
   const [isLoading, setIsLoading] = useState(false);
-  const router = useRouter();
-  const refreshData = async () => {
-    await router.replace(router.asPath);
-  };
+  const [err, setErr] = useState<string | undefined>(undefined);
+  const refetchInterval = 20000;
+
+  const { query } = useRouter();
+  const { gatewayUID } = query as GatewayDetailsQueryInterface;
+
+  const {
+    isLoading: gatewayLoading,
+    error: gatewayErr,
+    data: gateway,
+    refetch: gatewayRefetch,
+  } = useGateway(gatewayUID, refetchInterval);
```

Comment:
Does our API need to be as complicated as the useQuery API? (need to have all 4 of these functions?)